sql-plugin/src/main/340/scala/com/nvidia/spark/rapids/shims/GlobalLimitShims.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+import com.nvidia.spark.rapids.{RowCountPlanVisitor, SparkPlanMeta}
+
+import org.apache.spark.sql.execution.GlobalLimitExec
+
+object GlobalLimitShims {
+
+  /**
+   * Estimate the number of rows for a GlobalLimitExec.
+   */
+  def visit(plan: SparkPlanMeta[GlobalLimitExec]): Option[BigInt] = {
+    // offset is ignored now, but we should support it.
```

Comment:
We need to verify offset is zero and fallback to CPU if it is not, otherwise we silently do the wrong thing.

Reply:
Done

Reply:
> I thought this can be handled in a follow-up PR which is to fix https://github.com/NVIDIA/spark-rapids/issues/5589.

Supporting the feature does not preclude doing something safe when we don't support the feature.  The problem with punting it to that followup is that it's not clear that followup is a blocker to safely supporting Spark 3.4. Making the plugin safe for new features is the way to go until that feature is supported.